core/src/main/scala/akka/persistence/jdbc/journal/dao/JournalQueries.scala

```diff
@@ -23,20 +23,37 @@ class JournalQueries(
   private val insertAndReturn = JournalTable.returning(JournalTable.map(_.ordering))
   private val TagTableC = Compiled(TagTable)
 
-  def writeJournalRows(xs: Seq[(JournalAkkaSerializationRow, Set[String])])(implicit ec: ExecutionContext) = {
+  def writeJournalRows(xs: Seq[(JournalAkkaSerializationRow, Set[String])])(implicit ec: ExecutionContext): DBIOAction[Any, NoStream, Effect.Write] = {
     val sorted = xs.sortBy((event => event._1.sequenceNumber))
     if (sorted.exists(_._2.nonEmpty)) {
       // only if there are any tags
-      val (events, tags) = sorted.unzip
+      writeEventsAndTags(sorted)
+    } else {
+      // optimization avoid some work when not using tags
+      val events = sorted.map(_._1)
+      JournalTableC ++= events
+    }
+  }
+
+  private def writeEventsAndTags(sorted: Seq[(JournalAkkaSerializationRow, Set[String])])(
+      implicit ec: ExecutionContext): DBIOAction[Any, NoStream, Effect.Write] = {
+    val (events, tags) = sorted.unzip
+    // using revision by default.
+    val tagRows = sorted.map { case (eventRow, tags) =>
```

Comment:
I think it would be easier to understand if this is moved to the two `if` cases and then constructed with the eventId or not, instead of the additional map + copy